official-antistasi-community/A3-Antistasi
# Move to new templates

## What type of PR is this.
1. [ ] Bug
2. [ ] Change
3. [x] Enhancement

### What have you changed and why?
Information: The move to the new template system in the code base.
- Removed old template system files
- renamed newTemplates to just templates
- added macros for "ease" of use with the new system
- move compilation of groups and assets lists to the templates processing.
- moved faction data to hashmap from namespace
- re did the arty entry in templates to be like the other vehicles entries, added a magazines entry thats a hashmap matching vehicle to magazines
- added template validation to loadFaction, verifying loadouts and datatype of entries in the template (not all entries have verification yet, and bad templates havent been tested yet, validation fail only screams in the log)
- removed list of assets and groups from initVarServer to two new functions in the faction loading process.
- Added 2 Weapons to 3CB Faction Logistics, Added 1 Vehicle, Added Trucks to Coveredblacklist.
- Added Line to add Static Weapons to StaticsToSave


prob a lot more.

### Please specify which Issue this PR Resolves.
closes #XXXX

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [ ] No
2. [x] Yes (Please provide further detail below.)
 see bellow
### How can the changes be tested?
Steps: Lots of play testing

Check if Statics get added to StaticsToSave,
Check fitment, clipping with new added Statics and the Pickup Truck.
********************************************************
Notes: theres a lot of small stuff like rebel groups that are a bit odd, and should be changed still, but this is probably better for another PR as this is already pretty large.
